misk/src/main/kotlin/misk/hibernate/SchemaMigratorService.kt

```diff
@@ -0,0 +1,32 @@
+package misk.hibernate
+
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.inject.Key
+import misk.DependentService
+import misk.environment.Environment
+import misk.inject.toKey
+import javax.inject.Provider
+import kotlin.reflect.KClass
+
+internal class SchemaMigratorService(
+  private val environment: Environment,
+  qualifier: KClass<out Annotation>,
+  private val schemaMigratorProvider: Provider<SchemaMigrator> // Lazy!
+) : AbstractIdleService(), DependentService {
+
+  override val consumedKeys = setOf<Key<*>>(SessionFactoryService::class.toKey(qualifier))
+  override val producedKeys = setOf<Key<*>>(SchemaMigratorService::class.toKey(qualifier))
+
+  override fun startUp() {
+    val schemaMigrator = schemaMigratorProvider.get()
+    if (environment == Environment.TESTING || environment == Environment.DEVELOPMENT) {
```

Comment:
unrelated to your PR, but I think we'll see this condition a lot. we should add something to `Environment`

Reply:
Yeah, though we gotta be careful with words. `environment.isNonProduction()`!